andywiecko/plotex
# [WIP] Refactor/python setup

This prepares plotex to be distributed as a Python package. Changes in this PR:

- added setup.py,
- restructured directory tree so that it conforms to standard ways of organizing Python package
- switched from storing Profile configuration in .py files to storing it in yml files (accessed as package resources)
- engine for loading the aforementioned profile data.

To test:
- Uninstall the currently installed version of plotex
- Install this version using pip by issuing:
```
pip install .
```
In the top-level directory of the repository. If you are not using virtualenv you should append --user flag to pip options in the command above or install the package with root privileges (although this is not recommended).

Known issues:
- temporary files are now stored in Python specific site-packages directory. We can do this better,
but let's leave it for another PR.
